core/core-impl/src/main/java/ai/timefold/solver/core/config/solver/SolverConfigOverride.java

```diff
@@ -0,0 +1,51 @@
+package ai.timefold.solver.core.config.solver;
+
+import java.util.function.Consumer;
+
+import ai.timefold.solver.core.config.AbstractConfig;
+import ai.timefold.solver.core.config.solver.termination.TerminationConfig;
+import ai.timefold.solver.core.config.util.ConfigUtils;
+
+public class SolverConfigOverride extends AbstractConfig<SolverConfigOverride> {
```

Comment:
I wouldn't implement `AbstractConfig`. This class is not an actual config, it doesn't serialize to XML, it belongs to the `solver` package. For the same reason, I'd remove the getters and setters and just leave the `with` method; but you may need the getters later, so I'd at least make them package-private.

Let's not give people two ways of doing the same thing.